### ISSUE TITLE
Spec counter not registering first spec after weapon switch on same enemy

### ISSUE THREAD

**Issue by laagone** [runelite/runelite#11097]
As the title suggests, with today's change to the special attack counter plugin, it appears to sometimes not register the first special attack on the same enemy should it be successful; more specifically, with soloing Corp (wherein I do 2 DWH, then 20 Arclight, then 200 BGS damage), after switching from one weapon to another, the first special attack with that weapon doesn't always get added to the counter (though it sometimes does, I haven't noticed a pattern to this inconsistency). I'm guessing it does this because all the attacks are done consecutively on the same enemy.

**Comment by Kamielvf** [comment]
I was messing around with this and posted some results on discord, will leave them here as I think this is most likely also what is happening here (interacting changing)

```
so basically, with the spec counter currently you can change interacting before the special damage is applied
if you interact with nothing, it will fail to track that spec
if you interact with another npc, and hit that npc after, it will track damage you dealt on that npc as spec
so if you hit npc A with spec but click on npc B as you spec (on npc A), then hit npc B normally, it applies that damage as spec
thats the interacting issue

and then the issue where it doesn't count 0 as spec hit and will take the first red hitsplat instead
```

**Comment by Adam-** [comment]
fixed in 844a85392bdf77993e72e2997d8fe5e3a05ae6a8
